src/OmniSharp.Roslyn.CSharp/Services/RoslynGotoDefinition.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.Text;
+using OmniSharp.Endpoints;
+using OmniSharp.Models;
+
+namespace OmniSharp.Roslyn.CSharp.Services
+{
+    [Export(typeof(IGotoDefintion))]
+    class RoslynGotoDefinition : IGotoDefintion
+    {
+        public Task<bool> IsApplicableTo(GotoDefinitionRequest request)
+        {
+            return Task.FromResult(OmnisharpWorkspace.Instance.GetDocument(request.FileName) != null);
```

Comment:
The request object seems too heavyweight for this method. Should just be ```Language``` or ```Extension``` for file extension I think. I can't think of a situation where we need more than that.